Sources/PointFree/Account/Invoices.swift

```diff
@@ -0,0 +1,302 @@
+
+    return { conn in
+      let subscription = conn.data.first
+
+      return AppEnvironment.current.stripe.fetchInvoices(subscription.customer)
+        .withExcept(notifyError(subject: "Couldn't load invoices"))
+        .run
+        .flatMap {
+          switch $0 {
+          case let .right(invoices):
+            return conn.map(const(subscription .*. invoices .*. conn.data.second))
+              |> middleware
+          case .left:
+            return conn |> redirect(
+              to: .account(.index),
+              headersMiddleware: flash(
+                .error,
+                """
+                We had some trouble loading your invoices! Please try again later.
+                If the problem persists, please notify <support@pointfree.co>.
+                """
+              )
+            )
+          }
+      }
+    }
+}
+
+// MARK: Views
+
+let invoicesView = View<(Stripe.Subscription, Stripe.ListEnvelope<Stripe.Invoice>, Database.User)> { subscription, invoicesEnvelope, currentUser -> Node in
+
+  gridRow([
+    gridColumn(sizes: [.mobile: 12, .desktop: 8], [style(margin(leftRight: .auto))], [
+      div([`class`([Class.padding([.mobile: [.all: 3], .desktop: [.all: 4]])])],
+          titleRowView.view(unit)
+            <> invoicesRowView.view(invoicesEnvelope)
+      )
+      ])
+    ])
+}
+
+private let titleRowView = View<Prelude.Unit> { _ in
+  gridRow([`class`([Class.padding([.mobile: [.bottom: 2]])])], [
+    gridColumn(sizes: [.mobile: 12], [
+      div([
+        h1([`class`([Class.pf.type.title2])], ["Payment history"])
+        ])
+      ])
+    ])
+}
+
+private let invoicesRowView = View<Stripe.ListEnvelope<Stripe.Invoice>> { invoicesEnvelope in
+  div(
+    invoicesEnvelope.data.map { invoice in
+      gridRow([`class`([Class.padding([.mobile: [.bottom: 2]])])], [
+        gridColumn(sizes: [.mobile: 4], [`class`([Class.type.fontFamily.monospace])], [
+          div([text("#" + invoice.number.unwrap)])
+          ]),
+        gridColumn(sizes: [.mobile: 4], [`class`([Class.type.align.end, Class.type.fontFamily.monospace])], [
+          div([text(dateFormatter.string(from: invoice.date))])
+          ]),
+        gridColumn(sizes: [.mobile: 2], [`class`([Class.type.align.end, Class.type.fontFamily.monospace])], [
+          div([text(format(cents: invoice.total))])
+          ]),
+        gridColumn(sizes: [.mobile: 2], [`class`([Class.grid.end(.mobile), Class.grid.end(.desktop)])], [
+          div([
+            a(
+              [
+                `class`([Class.pf.components.button(color: .purple, size: .small)]),
+                href(path(to: .account(.invoices(.show(invoice.id))))),
+                target(.blank),
+              ],
+              ["Print"]
+            )
+            ])
+          ]),
+        ])
+    }
+  )
+}
+
+let invoiceView = View<(Stripe.Subscription, Database.User, Stripe.Invoice)> { subscription, currentUser, invoice -> Node in
```

Comment:
totally not necessary, but I could see this view being broken down into some smaller subviews. it's p big!

we could save it for another day.